features/steps/temporary.py

```diff
@@ -0,0 +1,126 @@
+import json
+from behave import Given, Then
+
+
+def _parse_destination_url_from_log(log: str) -> str:
+    """
+    Get the receiving path from the docker log
+    """
+    path_line = [x for x in log.split("\n") if "this-requests-url" in x]
+    assert len(path_line) == 1, f'Cannot find "this-requests-url" in service log {log}'
+    return "/" + path_line[0].split("/")[-1].strip()
+
+
+def _parse_request_headers_as_dict_from_log(log: str) -> dict:
+    """
+    Get the headers from the docker logs as a dictionary.
+    """
+    assert "this-requests-headers:" in log, (
+        "'this-requests-headers:' expected but not found in: " f"{log}"
+    )
+    headers_dict_as_str = (
+        log.split("this-requests-headers:")[1].replace("'", '"').split("}")[0] + "}"
+    )
+    headers_as_dict = json.loads(headers_dict_as_str)
+    return headers_as_dict
+
+
+def get_request_logs(container, request_id) -> str:
+    """
+    Get docker logs for this request as one big string.
+    """
+    # Get logs relevant to this specific request
+    docker_logs = str(container.logs().decode("utf-8"))
+    assert (
+        request_id in docker_logs
+    ), f"""
+        Could not find: {request_id} in:
+        {str(docker_logs)}
+        """
+    return docker_logs.split(request_id)[1]
+
+
+@Given('a request json payload of "{request_json_payload}"')
+def step_impl(context, request_json_payload):
+    """
+    Load a JSON payload file to submit as part of the request.
+    """
+    with open(request_json_payload, "r") as f:
+        context.json = json.load(f)
+
+
+@Given("a request with the headers")
+def step_impl(context):
+    """
+    Set request headers from table in scenario.
+    """
+    headers = {}
+    for row in context.table:
+        key = row["key"].strip()
+        value = row["value"].strip()
+        headers[key] = value
+    context.request_headers = headers
+
+
+@Given('I send the request to the upload service at "{upload_service_endpoint_url}"')
+def step_impl(context, upload_service_endpoint_url):
+    """
+    Submit a GET request to the specified endpoint.
+    """
+    context.request_path = upload_service_endpoint_url
+    context.response = context.session.get(
+        f"http://127.0.0.1:5001{upload_service_endpoint_url}",
+        headers=context.request_headers,
+        json=context.json,
+    )
+    context.receiving_service_log = get_request_logs(
+        context.backend_container, context.request_id
+    )
+
+
+@Then('the backend receives a request to "{upload_service_endpoint}"')
```

Comment:
can you make it `service_endpoint`, there's an api to come and we can reuse this if we make it nice and generic.